src/Func.cpp

```diff
@@ -164,9 +164,10 @@ void Func::define_extern(const std::string &function_name,
                          const std::vector<Type> &types,
                          int dimensionality,
                          NameMangling mangling,
-                         bool uses_old_buffer_t) {
+                         bool uses_old_buffer_t,
+                         DeviceAPI device_api) {
```

Comment:
Or add an overload. One takes just the bool and the other takes DeviceAPI and a bool.